app/controllers/documents_controller.rb

```diff
@@ -19,6 +20,14 @@ def edit
   def update
     @document = @planning_application.documents.find(params[:id])
     if @document.update(document_params)
+      if @document.saved_change_to_attribute?(:validated)
+        # we want to track cases where the document was changed from invalid to valid, hence this statement
+        if @document.saved_changes[:validated][1] == true && @document.saved_changes[:validated][0] == false
```

Comment:
Could these be helper methods that simply return true or false?

Reply:
I am a little afraid the code can get hard to read since we are using an "audit" helper method already here. on line 23 it is just a true/false that is being returned. Then on 25 we could abstract this but then we'd need to be going up and down to see what is being audited, and is it worth abstracting given its only used once? 

Not sure that would make this more unreadable? 🤔